sklearn/ensemble/iforest.py

```diff
@@ -314,7 +314,7 @@ def _average_path_length(n_samples_leaf):
 
         average_path_length[mask] = 1.
         average_path_length[not_mask] = 2. * (
-            np.log(n_samples_leaf[not_mask]) + 0.5772156649) - 2. * (
+            np.log(n_samples_leaf[not_mask]) + np.euler_gamma) - 2. * (
```

Comment:
I'd like to see a separate test for `average_path_length` testing equivalence between the integer and array cases. Please add.